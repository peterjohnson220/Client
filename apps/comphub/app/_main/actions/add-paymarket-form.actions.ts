import { Action } from '@ngrx/store';

import { AddPayMarketRequest } from 'libs/models/payfactors-api';
import { PricingPaymarket } from '../models';

export const OPEN_FORM = '[Comphub/Add PayMarket Form] Open Add Pay Market Form';
export const CLOSE_FORM = '[Comphub/Add PayMarket Form] Close Add Pay Market Form';
export const SAVE_PAYMARKET = '[Comphub/Add PayMarket Form] Save Pay Market';
export const SAVE_PAYMARKET_SUCCESS = '[Comphub/Add PayMarket Form] Save Pay Market Success';
export const SAVE_PAYMARKET_ERROR = '[Comphub/Add PayMarket Form] Save Pay Market Error';
export const SAVE_PAYMARKET_CONFLICT = '[Comphub/Add PayMarket Form] Save Pay Market Conflict';
export const CLEAR_SAVE_ERROR = '[Comphub/Add PayMarket Form] Clear Save Error';
export const GET_DISMISS_INFO_BANNER_SETTING = '[Comphub/Add PayMarket Form] Get Dismiss Info Banner Setting';
export const OPEN_INFO_BANNER = '[Comphub/Add PayMarket Form] Open Info Banner';
export const CLOSE_INFO_BANNER = '[Comphub/Add PayMarket Form] Close Info Banner';

export class OpenForm implements Action {
  readonly type = OPEN_FORM;

  constructor(public payload?: { showSkipButton: boolean }) {}
}

export class CloseForm implements Action {
  readonly type = CLOSE_FORM;
}

export class SavePaymarket implements Action {
  readonly type = SAVE_PAYMARKET;

  constructor(public payload: AddPayMarketRequest) {}
}

export class SavePaymarketSuccess implements Action {
  readonly type = SAVE_PAYMARKET_SUCCESS;

  constructor(public payload: PricingPaymarket) {}
}

export class SavePaymarketError implements Action {
  readonly type = SAVE_PAYMARKET_ERROR;
}

export class SavePaymarketConflict implements Action {
  readonly type = SAVE_PAYMARKET_CONFLICT;
}

export class ClearSaveError implements Action {
  readonly type = CLEAR_SAVE_ERROR;
}

export class GetDismissInfoBannerSetting implements Action {
  readonly type = GET_DISMISS_INFO_BANNER_SETTING;
}

export class OpenInfoBanner implements Action {
  readonly type = OPEN_INFO_BANNER;
}

export class CloseInfoBanner implements Action {
  readonly type = CLOSE_INFO_BANNER;
}

export type Actions
  = OpenForm
  | CloseForm
  | SavePaymarket
  | SavePaymarketSuccess
  | SavePaymarketError
  | SavePaymarketConflict
  | ClearSaveError
  | GetDismissInfoBannerSetting
  | OpenInfoBanner
  | CloseInfoBanner;
