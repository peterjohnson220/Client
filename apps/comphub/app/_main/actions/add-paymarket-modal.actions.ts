import { Action } from '@ngrx/store';

import { PayMarket } from 'libs/models';
import { AddPayMarketRequest } from 'libs/models/payfactors-api';

export const OPEN_MODAL = '[Comphub/Add PayMarket Modal] Open Add Pay Market Modal';
export const CLOSE_MODAL = '[Comphub/Add PayMarket Modal] Close Add Pay Market Modal';
export const SAVE_PAYMARKET = '[Comphub/Add PayMarket Modal] Save Pay Market';
export const SAVE_PAYMARKET_SUCCESS = '[Comphub/Add PayMarket Modal] Save Pay Market Success';
export const SAVE_PAYMARKET_ERROR = '[Comphub/Add PayMarket Modal] Save Pay Market Error';
export const SAVE_PAYMARKET_CONFLICT = '[Comphub/Add PayMarket Modal] Save Pay Market Conflict';
export const CLEAR_SAVE_ERROR = '[Comphub/Add PayMarket Modal] Clear Save Error';

export class Open implements Action {
  readonly type = OPEN_MODAL;
}

export class Close implements Action {
  readonly type = CLOSE_MODAL;
}

export class Save implements Action {
  readonly type = SAVE_PAYMARKET;

  constructor(public payload: AddPayMarketRequest) {}
}

export class SaveSuccess implements Action {
  readonly type = SAVE_PAYMARKET_SUCCESS;

  constructor(public payload: { companyPayMarketId: number }) {}
}

export class SaveError implements Action {
  readonly type = SAVE_PAYMARKET_ERROR;
}

export class SaveConflict implements Action {
  readonly type = SAVE_PAYMARKET_CONFLICT;
}

export class ClearSaveError implements Action {
  readonly type = CLEAR_SAVE_ERROR;
}

export type Actions
  = Open
  | Close
  | Save
  | SaveSuccess
  | SaveError
  | SaveConflict
  | ClearSaveError;
