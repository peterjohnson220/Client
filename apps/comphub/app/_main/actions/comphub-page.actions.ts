import { HttpErrorResponse } from '@angular/common/http';

import { Action } from '@ngrx/store';

import { ComphubPages } from '../data';
import { CountryDataSet, JobPricingLimitInfo, ExchangeDataSet, FooterContextRequest, FooterContext } from '../models';

export const INIT = '[Comphub/Comphub Page] Init Comphub Page';
export const NAVIGATE_TO_CARD = '[Comphub/Comphub Page] Navigate to Card';
export const NAVIGATE_TO_NEXT_CARD = '[Comphub/Comphub Page] Navigate to Next Card';
export const NAVIGATE_TO_PREVIOUS_CARD = '[Comphub/Comphub Page] Navigate to Previous Card';
export const ADD_ACCESSIBLE_PAGES = '[Comphub/Comphub Page] Add Accessible Pages';
export const REMOVE_ACCESSIBLE_PAGES = '[Comphub/Comphub Page] Remove Accessible Pages';
export const RESET_ACCESSIBLE_PAGES = '[Comphub/Comphub Page] Reset Accessible Pages';
export const RESET_PAGES_ACCESSED = '[Comphub/Comphub Page] Reset Pages Accessed';
export const UPDATE_CARD_SUBTITLE = '[Comphub/Comphub Page] Update Card Subtitle';
export const GET_JOB_PRICING_LIMIT_INFO = '[Comphub/Comphub Page] Get Job Pricing Limit Info';
export const SET_JOB_PRICING_LIMIT_INFO = '[Comphub/Comphub Page] Set Job Pricing Limit Info';
export const GET_COUNTRY_DATA_SETS = '[Comphub/Comphub Page] Get Active Country Data Sets';
export const GET_COUNTRY_DATA_SETS_SUCCESS = '[Comphub/Comphub Page] Get Country Data Sets Success';
export const HANDLE_API_ERROR = '[Comphub/Comphub Page] Handle API Error';
export const UPDATE_ACTIVE_COUNTRY_DATA_SET = '[Comphub/Comphub Page] Update Active Country Data Set';
export const GET_EXCHANGE_DATA_SETS = '[Comphub/Comphub Page] Get Exchange Data Sets';
export const GET_EXCHANGE_DATA_SETS_SUCCESS = '[Comphub/Comphub Page] Get Exchange Data Sets Success';
export const UPDATE_ACTIVE_EXCHANGE_DATA_SET = '[Comphub/Comphub Page] Update Active Exchange Data Set';
export const SET_QUICK_PRICE_TYPE_IN_WORKFLOW_CONTEXT = '[Comphub/Comphub Page] Set Quick Price Type In Workflow Context';
export const SET_QUICK_PRICE_HISTORY_MODAL_OPEN = '[Comphub/Comphub Page] Set Quick Price History Modal Open';
export const UPDATE_FOOTER_CONTEXT = '[Comphub/Comphub Page] Update Footer Context';
export const SET_FOOTER_CONTEXT = '[Comphub/Comphub Page] Set Footer Context';

export class Init implements Action {
  readonly type = INIT;

  constructor() {}
}

export class NavigateToCard implements Action {
  readonly type = NAVIGATE_TO_CARD;

  constructor(public payload: { cardId: string }) {}
}

export class NavigateToNextCard implements Action {
  readonly type = NAVIGATE_TO_NEXT_CARD;

  constructor() {}
}

export class NavigateToPreviousCard implements Action {
  readonly type = NAVIGATE_TO_PREVIOUS_CARD;

  constructor() {}
}

export class AddAccessiblePages implements Action {
  readonly type = ADD_ACCESSIBLE_PAGES;

  constructor(public payload: ComphubPages[]) {}
}

export class RemoveAccessiblePages implements Action {
  readonly type = REMOVE_ACCESSIBLE_PAGES;

  constructor(public payload: ComphubPages[]) {}
}

export class ResetAccessiblePages implements Action {
  readonly type = RESET_ACCESSIBLE_PAGES;

  constructor() {}
}

export class ResetPagesAccessed implements Action {
  readonly type = RESET_PAGES_ACCESSED;

  constructor() {}
}

export class UpdateCardSubtitle implements Action {
  readonly type = UPDATE_CARD_SUBTITLE;

  constructor(public payload: { cardId: ComphubPages, subTitle: string }) {}
}

export class GetJobPricingLimitInfo implements Action {
  readonly type = GET_JOB_PRICING_LIMIT_INFO;

  constructor() {}
}

export class SetJobPricingLimitInfo implements Action {
  readonly type = SET_JOB_PRICING_LIMIT_INFO;

  constructor(public payload: JobPricingLimitInfo) {}
}

export class GetCountryDataSets implements Action {
  readonly type = GET_COUNTRY_DATA_SETS;
}

export class GetCountryDataSetsSuccess implements Action {
  readonly type = GET_COUNTRY_DATA_SETS_SUCCESS;

  constructor(public payload: CountryDataSet[]) { }
}

export class UpdateActiveCountryDataSet implements Action {
  readonly type = UPDATE_ACTIVE_COUNTRY_DATA_SET;

  constructor(public payload: string) { }
}

export class HandleApiError implements Action {
readonly type = HANDLE_API_ERROR;

  constructor(public payload: HttpErrorResponse) {}
}

export class GetExchangeDataSets implements Action {
  readonly type = GET_EXCHANGE_DATA_SETS;
}

export class GetExchangeDataSetsSuccess implements Action {
  readonly type = GET_EXCHANGE_DATA_SETS_SUCCESS;

  constructor(public payload: ExchangeDataSet[]) { }
}

export class UpdateActiveExchangeDataSet implements Action {
  readonly type = UPDATE_ACTIVE_EXCHANGE_DATA_SET ;

  constructor(public payload: number) { }
}

export class SetQuickPriceTypeInWorkflowContext implements Action {
  readonly type = SET_QUICK_PRICE_TYPE_IN_WORKFLOW_CONTEXT;

  constructor(public payload: string) {}
}

export class SetQuickPriceHistoryModalOpen implements Action {
  readonly type = SET_QUICK_PRICE_HISTORY_MODAL_OPEN;
  constructor(public isOpen: boolean) {}
}

export class UpdateFooterContext implements Action {
  readonly type = UPDATE_FOOTER_CONTEXT;
  constructor() {}
}

export class SetFooterContext implements Action {
  readonly type = SET_FOOTER_CONTEXT;
  constructor(public payload: FooterContext) {}
}

export type Actions
  = Init
  | NavigateToCard
  | NavigateToNextCard
  | NavigateToPreviousCard
  | AddAccessiblePages
  | RemoveAccessiblePages
  | ResetAccessiblePages
  | ResetPagesAccessed
  | UpdateCardSubtitle
  | GetJobPricingLimitInfo
  | SetJobPricingLimitInfo
  | GetCountryDataSets
  | GetCountryDataSetsSuccess
  | UpdateActiveCountryDataSet
  | HandleApiError
  | GetExchangeDataSets
  | GetExchangeDataSetsSuccess
  | UpdateActiveExchangeDataSet
  | SetQuickPriceTypeInWorkflowContext
  | SetQuickPriceHistoryModalOpen
  | UpdateFooterContext
  | SetFooterContext;
