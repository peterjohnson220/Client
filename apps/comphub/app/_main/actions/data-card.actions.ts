import { Action } from '@ngrx/store';

import { JobData, JobGridData, QuickPriceGridContext } from '../models';
import { RateType } from '../data';

export const CARD_OPENED  = '[Comphub/Data Card] Card Opened';
export const GET_QUICK_PRICE_MARKET_DATA  = '[Comphub/Data Card] Get quick Price Data';
export const GET_QUICK_PRICE_MARKET_DATA_SUCCESS  = '[Comphub/Data Card] Get quick Price Data Success';
export const GET_QUICK_PRICE_MARKET_DATA_ERROR  = '[Comphub/Data Card] Get quick Price Data Error';
export const SET_SELECTED_JOB_DATA  = '[Comphub/Data Card] Set Selected Job Data';
export const SET_SELECTED_RATE = '[Comphub/Data Card] Set Selected Rate';
export const CLEAR_SELECTED_JOB_DATA  = '[Comphub/Data Card] Clear Selected Job Data';
export const SET_MARKET_DATA_CHANGE = '[Comphub/Data Card] Set Market Data Change';
export const SHOW_PEER_BANNER = '[Comphub/Data Card] Show Peer Banner';
export const TOGGLE_JOB_DESCRIPTION = '[Comphub/Data Card] Toggle Job Description';

export class CardOpened implements Action {
  readonly type = CARD_OPENED;
}

export class GetQuickPriceMarketData implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA;

  constructor(public payload: QuickPriceGridContext) {}
}

export class GetQuickPriceMarketDataSuccess implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA_SUCCESS;

  constructor(public payload: JobGridData) {}
}

export class GetQuickPriceMarketDataError implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA_ERROR;
}

export class SetSelectedJobData implements Action {
  readonly type = SET_SELECTED_JOB_DATA;

  constructor(public payload: JobData) {}
}

export class SetSelectedRate implements Action {
  readonly type = SET_SELECTED_RATE;

  constructor(public payload: RateType) {}
}

export class ClearSelectedJobData implements Action {
  readonly type = CLEAR_SELECTED_JOB_DATA;

  constructor() {}
}

export class SetMarketDataChange implements Action {
  readonly type = SET_MARKET_DATA_CHANGE;

  constructor(public payload: boolean) {}
}

export class ShowPeerBanner implements Action {
  readonly type = SHOW_PEER_BANNER;

  constructor() {}
}

export class ToggleJobDescription implements Action {
  readonly type = TOGGLE_JOB_DESCRIPTION;

  constructor(public payload: { jobId: number }) {}
}

export type Actions
  = CardOpened
  | GetQuickPriceMarketData
  | GetQuickPriceMarketDataSuccess
  | GetQuickPriceMarketDataError
  | SetSelectedJobData
  | SetSelectedRate
  | ClearSelectedJobData
  | SetMarketDataChange
  | ShowPeerBanner
  | ToggleJobDescription;
