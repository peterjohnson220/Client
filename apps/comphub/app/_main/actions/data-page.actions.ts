import { Action } from '@ngrx/store';
import { GridDataResult } from '@progress/kendo-angular-grid';

import { QuickPriceGridContext } from '../models';

export const GET_QUICK_PRICE_MARKET_DATA  = '[Comphub/Data Page] Get quick Price Data';
export const GET_QUICK_PRICE_MARKET_DATA_SUCCESS  = '[Comphub/Data Page] Get quick Price Data Success';
export const GET_QUICK_PRICE_MARKET_DATA_ERROR  = '[Comphub/Data Page] Get quick Price Data Error';

export class GetQuickPriceMarketData implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA;

  constructor(public payload: QuickPriceGridContext) {}
}

export class GetQuickPriceMarketDataSuccess implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA_SUCCESS;

  constructor(public payload: GridDataResult) {}
}

export class GetQuickPriceMarketDataError implements Action {
  readonly type = GET_QUICK_PRICE_MARKET_DATA_ERROR;
}

export type Actions
  = GetQuickPriceMarketData
  | GetQuickPriceMarketDataSuccess
  | GetQuickPriceMarketDataError;
