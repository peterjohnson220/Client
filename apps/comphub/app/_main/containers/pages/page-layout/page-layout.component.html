<div class="comphub-page">
  <div class="page-header">
    <i class="fa {{pageIconClass}} page-icon" aria-hidden="true"></i>
    <span class="page-title">{{pageTitle}}</span>
  </div>
  <div>{{pageSubTitle}}</div>

  <ng-content select="[page-content]"></ng-content>

  <ng-content select="[actions-section]"></ng-content>

</div>
