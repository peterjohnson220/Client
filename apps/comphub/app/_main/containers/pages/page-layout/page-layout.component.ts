import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'pf-comphub-page-layout',
  templateUrl: './page-layout.component.html',
  styleUrls: ['./page-layout.component.scss']
})
export class PageLayoutComponent implements OnInit {
  @Input() pageTitle = '';
  @Input() pageSubTitle = '';
  @Input() pageIconClass = '';

  constructor() { }

  ngOnInit() {
  }

}
