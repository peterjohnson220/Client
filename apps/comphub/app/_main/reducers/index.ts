import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromJobsCardReducer from './jobs-card.reducer';
import * as fromComphubPageReducer from './comphub-page.reducer';
import * as fromAddPayMarketFormReducer from './add-paymarket-form.reducer';
import * as fromMarketsCardReducer from './markets-card.reducer';
import * as fromDataCardReducer from './data-card.reducer';
import * as fromSummaryCardReducer from './summary-card.reducer';

// Feature area state
export interface ComphubMainState {
  jobsCard: fromJobsCardReducer.State;
  comphubPage: fromComphubPageReducer.State;
  addPayMarketForm: fromAddPayMarketFormReducer.State;
  marketsCard: fromMarketsCardReducer.State;
  dataCard: fromDataCardReducer.State;
  summaryCard: fromSummaryCardReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  comphub_main: ComphubMainState;
}

// Feature area reducers
export const reducers = {
  jobsCard: fromJobsCardReducer.reducer,
  comphubPage: fromComphubPageReducer.reducer,
  marketsCard: fromMarketsCardReducer.reducer,
  dataCard: fromDataCardReducer.reducer,
  addPayMarketForm: fromAddPayMarketFormReducer.reducer,
  summaryCard: fromSummaryCardReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<ComphubMainState>('comphub_main');

// Feature Selectors
export const selectJobsCardState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.jobsCard
);

export const selectComphubPageState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.comphubPage
);

export const selectAddPayMarketFormState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.addPayMarketForm
);

export const selectMarketsCardState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.marketsCard
);

export const selectDataCardState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.dataCard
);

export const selectSummaryCardState = createSelector(
  selectFeatureAreaState,
  (state: ComphubMainState) => state.summaryCard
);

// Jobs Card
export const getTrendingJobGroups = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getTrendingJobGroups
);

export const getLoadingTrendingJobs = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getLoadingTrendingJobs
);

export const getLoadingTrendingJobsError = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getLoadingTrendingJobsError
);

export const getJobSearchOptions = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getJobSearchOptions
);

export const getExchangeJobSearchOptions = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getExchangeJobSearchOptions
);

export const getLoadingJobSearchOptions = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getLoadingJobSearchOptions
);

export const getLoadingJobSearchOptionsError = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getLoadingJobSearchOptionsError
);

export const getSelectedJob = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getSelectedJob
);

export const getSelectedExchangeJobId = createSelector(
  selectJobsCardState,
  fromJobsCardReducer.getSelectedExchangeJobId
);

// Comphub Page
export const getCards = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getCards
);

export const getSelectedPageId = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getSelectedPageId
);

export const getEnabledPages = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getEnabledPages
);

export const getPagesAccessed = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getPagesAccessed
);

export const getJobPricingLimitInfo = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getJobPricingLimitInfo
);

export const getJobPricingBlocked = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getJobPricingBlocked
);

export const getCountryDataSetsLoaded = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getCountryDataSetsLoaded
);

export const getCountryDataSets = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getCountryDataSets
);

export const getExchangeDataSets = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getExchangeDataSets
);

export const getActiveCountryDataSet = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getActiveCountryDataSet
);

export const getActiveExchangeDataSet = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getActiveExchangeDataSet
);

export const getWorkflowContext = createSelector(
  selectComphubPageState,
  fromComphubPageReducer.getWorkflowContext
);

// Markets Card
export const getSelectedPaymarket = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getSelectedPaymarket
);

export const getLoadingPaymarkets = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getLoadingPaymarkets
);

export const getLoadingPaymarketsError = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getLoadingPaymarketsError
);

export const getPaymarkets = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getPaymarkets
);

export const getPaymarketsFilter = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getPaymarketsFilter
);

export const getLoadingMarketDataLocations = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getLoadingMarketDataLocations
);

export const getMarketDataLocations = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getMarketDataLocations
);

export const getVisiblePaymarkets = createSelector(
  getPaymarkets, getPaymarketsFilter, (paymarkets, filter) => {
    return paymarkets.filter(x => !filter || x.PayMarketName.toLowerCase().indexOf(filter.toLowerCase()) !== -1);
  }
);

export const getMarketDataScope = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getMarketDataScope
);

export const getHideNewPaymarketsButton = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getHideNewPaymarketsButton
);

export const getMarketDataScopesLoading = createSelector(
  selectMarketsCardState,
  fromMarketsCardReducer.getLoadingMarketDataScopes
);

// Data Card
export const getJobGridResults = createSelector(
  selectDataCardState,
  fromDataCardReducer.getJobGridResults
);

export const getLoadingJobGridResults = createSelector(
  selectDataCardState,
  fromDataCardReducer.getLoadingJobGridResults
);

export const getLoadingJobGridResultsError = createSelector(
  selectDataCardState,
  fromDataCardReducer.getLoadingJobGridResultsError
);

export const getSelectedJobData = createSelector(
  selectDataCardState,
  fromDataCardReducer.getSelectedJobData
);

export const getMarketDataChange = createSelector(
  selectDataCardState,
  fromDataCardReducer.getMarketDataChange
);

export const getPeerBannerOpen = createSelector(
  selectDataCardState,
  fromDataCardReducer.getPeerBannerOpen
);

export const getSelectedRate = createSelector(
  selectDataCardState,
  fromDataCardReducer.getSelectedRate
);

export const getForcePeerMapRefresh = createSelector(
  selectDataCardState,
  fromDataCardReducer.getForcePeerMapRefresh
);

// Add Pay Market Form
export const getAddPayMarketFormOpen = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getFormOpen
);

export const getSavingPayMarket = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getSaving
);

export const getSavingPayMarketConflict = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getSavingConflict
);

export const getSavingPayMarketError = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getSavingError
);

export const getInfoBannerOpen = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getInfoBannerOpen
);

export const getShowSkipButton = createSelector(
  selectAddPayMarketFormState,
  fromAddPayMarketFormReducer.getShowSkipButton
);

// Summary Card
export const getSalaryTrendData = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getSalaryTrendData
);

export const getLoadingSalaryTrendData = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getLoadingSalaryTrendData
);

export const getLoadingSalaryTrendDataError = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getLoadingSalaryTrendError
);

export const getSharePricingSummaryModalOpen = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getSharePricingSummaryModalOpen
);

export const getSharePricingSummaryError = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getSharePricingSummaryError
);

export const getSharePricingSummaryConflict = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getSharePricingSummaryConflict
);

export const getCreatingProject = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getCreatingProject
);

export const getCreatingProjectError = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getCreatingProjectError
);

export const getCanAccessProjectsTile = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getCanAccessProjectTile
);

export const getGlossaryOpen = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getGlossaryOpen
);

export const getMinPaymarketMinimumWage = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getMinPaymarketMinimumWage
);

export const getMaxPaymarketMinimumWage = createSelector(
  selectSummaryCardState,
  fromSummaryCardReducer.getMaxPaymarketMinimumWage
);
