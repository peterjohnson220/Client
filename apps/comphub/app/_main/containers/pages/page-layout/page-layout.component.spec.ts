import { NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { combineReducers, Store, StoreModule } from '@ngrx/store';

import * as fromRootState from 'libs/state/state';

import * as fromComphubPageActions from '../../../actions/comphub-page.actions';
import * as fromComphubMainReducer from '../../../reducers';
import { PageLayoutComponent } from './page-layout.component';

describe('Comphub - Main - Page Layout', () => {
  let instance: PageLayoutComponent;
  let fixture: ComponentFixture<PageLayoutComponent>;
  let store: Store<fromComphubMainReducer.State>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          comphub_main: combineReducers(fromComphubMainReducer.reducers),
        })
      ],
      declarations: [ PageLayoutComponent ],
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    fixture = TestBed.createComponent(PageLayoutComponent);
    instance = fixture.componentInstance;

    store = TestBed.get(Store);
    fixture.detectChanges();
  });

  it('should dispatch a NavigateToNextCard action, when handling the next button click', () => {
    spyOn(store, 'dispatch');
    const expectedAction = new fromComphubPageActions.NavigateToNextCard();

    instance.handleNextButtonClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should dispatch a NavigateToPreviousCard action, when handling the back button click', () => {
    spyOn(store, 'dispatch');
    const expectedAction = new fromComphubPageActions.NavigateToPreviousCard();

    instance.handleBackButtonClicked();

    expect(store.dispatch).toHaveBeenCalledWith(expectedAction);
  });

  it('should hide the back button, when told to', () => {
    instance.hideBackButton = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should hide the next button, when told to', () => {
    instance.hideNextButton = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should disable next button, when told to', () => {
    instance.hideNextButton = false;
    instance.nextButtonEnabled = false;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
