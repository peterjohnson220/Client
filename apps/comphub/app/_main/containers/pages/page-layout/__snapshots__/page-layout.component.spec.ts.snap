// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comphub - Main - Page Layout should disable next button, when told to 1`] = `
<pf-comphub-page-layout
  hideNextButton="false"
  nextButtonEnabled="false"
  pageIconClass=""
  pageSubTitle=""
  pageTitle=""
  store={[Function Store]}
>
  <div
    class="comphub-page"
  >
    <div
      class="page-header"
    >
      <i
        aria-hidden="true"
        class="fa  page-icon"
      />
      <span
        class="page-title"
      >
        
      </span>
    </div>
    <div>
      
    </div>
    <div
      class="actions-section"
    >
      
      <button
        class="btn comphub-secondary-btn"
        data-qa-id="btn-back"
      >
         Back 
      </button>
      
      <button
        class="btn comphub-primary-btn"
        data-qa-id="btn-next"
        disabled=""
      >
         Next 
      </button>
    </div>
  </div>
</pf-comphub-page-layout>
`;

exports[`Comphub - Main - Page Layout should hide the back button, when told to 1`] = `
<pf-comphub-page-layout
  hideBackButton={[Function Boolean]}
  pageIconClass=""
  pageSubTitle=""
  pageTitle=""
  store={[Function Store]}
>
  <div
    class="comphub-page"
  >
    <div
      class="page-header"
    >
      <i
        aria-hidden="true"
        class="fa  page-icon"
      />
      <span
        class="page-title"
      >
        
      </span>
    </div>
    <div>
      
    </div>
    <div
      class="actions-section"
    >
      
      
      <button
        class="btn comphub-primary-btn"
        data-qa-id="btn-next"
        disabled=""
      >
         Next 
      </button>
    </div>
  </div>
</pf-comphub-page-layout>
`;

exports[`Comphub - Main - Page Layout should hide the next button, when told to 1`] = `
<pf-comphub-page-layout
  hideNextButton={[Function Boolean]}
  pageIconClass=""
  pageSubTitle=""
  pageTitle=""
  store={[Function Store]}
>
  <div
    class="comphub-page"
  >
    <div
      class="page-header"
    >
      <i
        aria-hidden="true"
        class="fa  page-icon"
      />
      <span
        class="page-title"
      >
        
      </span>
    </div>
    <div>
      
    </div>
    <div
      class="actions-section"
    >
      
      <button
        class="btn comphub-secondary-btn"
        data-qa-id="btn-back"
      >
         Back 
      </button>
      
    </div>
  </div>
</pf-comphub-page-layout>
`;
